lib/linter/linter.js

```diff
@@ -432,10 +433,22 @@ function getDirectiveComments(filename, ast, ruleMapper, warnInlineConfig) {
 
 /**
  * Normalize ECMAScript version from the initial config
- * @param  {number} ecmaVersion ECMAScript version from the initial config
+ * @param {Parser} parser The parser which uses this options.
+ * @param {number} ecmaVersion ECMAScript version from the initial config
  * @returns {number} normalized ECMAScript version
  */
-function normalizeEcmaVersion(ecmaVersion) {
+function normalizeEcmaVersion(parser, ecmaVersion) {
+
+    // RuleTester always wraps parser, parser.$parser is the original parser object
+    if ((parser.$parser || parser) === espree) {
```

Comment:
`$parser`: maybe worth a better name?

Reply:
Do we need this check anymore? Espree’s default will be 5 automatically.

Reply:
it's no harm to explicitly set ecma version to 5(if not specified). In case we want to change espree's defaults in later releases. :)

Reply:
I’d rather not have unneeded code. We also won’t be using this with the new config system, so can we just keep the changes related to normalizing “latest”?

Reply:
well, will update.

Reply:
to clarify: no need to normalize `undefined` to 5, but we still have to keep the check, as some 3rd-party parsers may have a different "latest".

Reply:
Just so I understand: you are saying if, for instance, the parser is "@babel/eslint-parser" and `parserOptions.ecmaVersion` is `"latest"`, then we won't make any changes?

Reply:
yes!

Reply:
Okay, makes sense. I think we are on the same page now. Thanks for your patience.